src/Agent.Worker/ContainerOperationProvider.cs

```diff
@@ -411,6 +411,14 @@ private async Task StartContainerAsync(IExecutionContext executionContext, Conta
                 }
             }
 
+            // Build JSON to expose docker container name mapping to env
+            var containerEnv = executionContext.Variables.Get(Constants.Variables.Agent.ContainerMapping);
+            var containerMapping = containerEnv != null ? JObject.Parse(containerEnv) : new JObject();
+            var containerInfo = new JObject();
+            containerInfo["id"] = container.ContainerId;
+            containerMapping[container.ContainerName] = containerInfo;
+            executionContext.Variables.Set(Constants.Variables.Agent.ContainerMapping, containerMapping.ToString());
```

Comment:
`StartContainerAsync` mutates the Container to set the ID, so I think it'd be cleaner to move this up to `StartContainersAsync` , so you can build the JObject once, add to it in the loop after each call to `StartContainerAsync` and avoid parsing the variable.

Reply:
Done